src/gateway/bridge/shard_manager.rs

```diff
@@ -182,34 +166,19 @@ impl ShardManager {
     /// This will communicate shard boots with the [`ShardQueuer`] so that they are properly
     /// queued.
     #[instrument(skip(self))]
-    pub fn initialize(&self) -> Result<()> {
-        let shard_index = self.shard_index.load(Ordering::Relaxed);
-        let shard_init = self.shard_init.load(Ordering::Relaxed);
-        let shard_total = self.shard_total.load(Ordering::Relaxed);
-
+    pub fn initialize(&self, shard_index: u32, shard_init: u32, shard_total: u32) -> Result<()> {
         let shard_to = shard_index + shard_init;
 
         for shard_id in shard_index..shard_to {
-            self.boot([ShardId(shard_id), ShardId(shard_total)]);
+            self.boot(ShardInfo {
+                id: ShardId(shard_id),
+                total: shard_total,
+            });
         }
 
         Ok(())
     }
 
-    /// Sets the new sharding information for the manager.
-    ///
-    /// This will shutdown all existing shards.
-    ///
-    /// This will _not_ instantiate the new shards.
-    #[instrument(skip(self))]
-    pub async fn set_shards(&self, index: u32, init: u32, total: u32) {
-        self.shutdown_all().await;
-
```

Comment:
Strange, I wonder why this line was even included if the only time this gets called is when the `ShardManager` is first instantiated. Is there some sort of control flow related to the channels/message passing that might reach this function through some other means?

Reply:
Seeing that I removed no call of this function other than the initial startup, it must have been removed at some point.